grammar.js

```diff
@@ -1339,6 +1448,146 @@ module.exports = grammar({
       $.identifier,
     ),
 
+    alter_database: $ => seq(
+      $.keyword_alter,
+      $.keyword_database,
+      $.identifier,
+      optional($.keyword_with),
+      choice(
+        seq($.rename_object),
+        seq($.change_ownership),
+        seq(
+          $.keyword_reset,
+          choice(
+            $.keyword_all,
+            field("configuration_parameter", $.identifier)
+          ),
+        ),
+        seq(
+          $.keyword_set,
+          choice(
+            seq($.keyword_tablespace, $.identifier),
+            seq(
+              field("configuration_parameter", $.identifier),
+              choice(
+                seq($.keyword_from, $.keyword_current),
+                $.set_configuration,
+              )
+            ),
+          ),
+        ),
+      ),
+    ),
+
+    alter_role: $ => seq(
+      $.keyword_alter,
+      choice(
+        $.keyword_role,
+        $.keyword_group,
+        $.keyword_user,
+      ),
+      choice($.identifier, $.keyword_all),
+      choice(
+        $.rename_object,
+        seq(optional($.keyword_with),repeat($._role_options)),
+        seq(
+          optional(seq($.keyword_in, $.keyword_database, $.identifier)),
+          choice(
+            seq(
+              $.keyword_set,
+              field("option", $.identifier),
+              choice(
+                seq($.keyword_from, $.keyword_current),
+                $.set_configuration,
+              ),
+            ),
+            seq(
+              $.keyword_reset,
+              choice(
+                $.keyword_all,
+                field("option", $.identifier),
+              )),
+          ),
+        )
+      ),
+    ),
+
+    set_configuration: $ => seq(
```

Comment:
I think this could be consolidated further -- not all the way up to `keyword_set` unfortunately but including the field (picking one of option/configuration_parameter) and the 'from current'